### PR TITLE
(RHINENG-12028): serialize_host refactoring

### PR DESCRIPTION
**Pull request by strider**
# Overview

This PR is being created to address [RHINENG-12028](https://issues.redhat.com/browse/RHINENG-12028).

See my comment with all the explanations about this refactoring and improvements that can be done, in the Jira ticket above or in the commit message.

## PR Checklist

- [ ] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Include raw query examples in the PR description, if adding/modifying SQL query
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: app/serialization.py [hunk 1]
```diff
@@ -111,109 +111,98 @@ def serialize_host(
...
-        stale_warning_timestamp = staleness_timestamps.stale_warning_timestamp(
-            host.modified_on, staleness["conventional_time_to_stale_warning"]
-        )
-        culled_timestamp = staleness_timestamps.culled_timestamp(
-            host.modified_on, staleness["conventional_time_to_delete"]
+    # Ensure additional_fields is a tuple
+    additional_fields = additional_fields or tuple()
+
+    # Determine staleness timestamps
+    def get_staleness_timestamps(host, staleness_timestamps, staleness):
```
Comment (jpramos123) [inline app/serialization.py:122]: If you can, move this function to the global scope, as I'm going to reuse it in another feature

**Review by jpramos123 — CHANGES_REQUESTED** [review]
LGTM, just need the `get_staleness_timestamps` to be move to global scope.

File: app/serialization.py [hunk 2]
```diff
@@ -111,109 +111,98 @@ def serialize_host(
...
+        """Helper function to calculate staleness timestamps based on host type."""
+        # import ipdb; ipdb.set_trace()
```
Comment (jpramos123) [inline app/serialization.py:124]:
```suggestion
```

**Comment by kruai** [comment]
I agree with @jpramos123's comments, but I think the code looks good beyond that. Good work!

File: app/serialization.py (hunk 1 above)
Comment (thearifismail) [reply app/serialization.py:122]: > If you can, move this function to the global scope, as I'm going to reuse it in another feature

@jpramos123 was this function not moved by PR #2123?

**Comment by strider** [comment]
Thanks @thearifismail, @kruai, and @jpramos123 for your comments.

I'll address that asap.
